cpp/meson.build

```diff
@@ -24,7 +24,7 @@ project(
     meson_version: '>=1.3.0',
     default_options: [
         'buildtype=release',
-        'c_std=c99',
+        'c_std=gnu11,c11',
```

Comment:
I bumped this to C11 as that is what the CMake configuration uses, but it also appears that the c-ares dependency requires some POSIX extensions

Reply:
Namely it appears that c-ares wants to include macros from <net/if.h> and looks for macros like CLOCK_MONOTONIC from time.h on non-windows platforms